sources/updater/update_metadata/src/lib.rs

```diff
@@ -419,35 +450,207 @@ pub fn load_manifest<T: tough::Transport>(repository: &tough::Repository<T>) ->
     .context(error::ManifestParse)
 }
 
-#[test]
-fn test_migrations_forward() {
-    // A manifest with four migration tuples starting at 1.0 and ending at 1.3.
-    // There is a shortcut from 1.1 to 1.3, skipping 1.2
-    let path = "./tests/data/migrations.json";
-    let manifest: Manifest = serde_json::from_reader(File::open(path).unwrap()).unwrap();
-    let from = Version::parse("1.0.0").unwrap();
-    let to = Version::parse("1.5.0").unwrap();
-    let targets = find_migrations(&from, &to, &manifest).unwrap();
-
-    assert!(targets.len() == 3);
-    let mut i = targets.iter();
-    assert!(i.next().unwrap() == "migration_1.1.0_a");
-    assert!(i.next().unwrap() == "migration_1.1.0_b");
-    assert!(i.next().unwrap() == "migration_1.5.0_shortcut");
-}
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use chrono::{DateTime, Duration, NaiveDate, Utc};
+
+    #[test]
+    fn test_update_ready() {
+        let mut waves = BTreeMap::new();
+        // Construct a DateTime object for 1/1/2000 00:00:00
+        let time = DateTime::<Utc>::from_utc(
+            NaiveDate::from_ymd(2000, 1, 1).and_hms_milli(0, 0, 0, 0),
+            Utc,
+        );
+        // One single wave (0th wave does not count) for every update that spans over 2048 millisecond,
+        // Each seed will be mapped to a single millisecond within this wave,
+        // e.g. seed 1 -> update is ready 1 millisecond past start of wave
+        // seed 500 -> update is ready 500 millisecond past start of wave, etc
+        waves.insert(0, time);
+        waves.insert(MAX_SEED, time + Duration::milliseconds(MAX_SEED as i64));
+        let mut update = Update {
+            variant: String::from("bottlerocket"),
+            arch: String::from("test"),
+            version: Version::parse("1.1.1").unwrap(),
+            max_version: Version::parse("1.1.1").unwrap(),
+            waves,
+            images: Images {
+                boot: String::from("boot"),
+                root: String::from("root"),
+                hash: String::from("hash"),
+            },
+        };
+
+        for seed in (100..500).step_by(100) {
+            assert!(
+                !update.update_ready(seed, time),
```

Comment:
```suggestion
                !update.update_ready(seed, time + Duration::milliseconds((seed as i64) - 1)),
```
A bit more specific, proving we become available at the exact time we intend...